### PR TITLE
Cache tags adapter

### PR DESCRIPTION
**Pull request by Mitch391**

### CODE REVIEW

File: tests/unit/ObjectBridgeService/CacheTagsAdapterTest.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        $this->adapter = new CacheTagsAdapter();
+        $this->elementInterfaceMock = $this->createMock(ElementInterface::class);
+        $this->elementInterfaceMock->method("getCacheTag")->willReturn("type_1");
+        $this->elementInterfaceMock->method("getCacheTags")->willReturn(["type_2" => "type_2"]);
+    }
+
+    public function testGetCacheTagsNoElementObjectsGivenReturnsEmptyArray(): void
+    {
+        $result = $this->adapter->getCacheTagsForEveryObject([], ...[]);
```
Comment (JeroenVanDerLaan) [inline tests/unit/ObjectBridgeService/CacheTagsAdapterTest.php]: can use `$this->assertEmpty($result);`

File: tests/unit/ObjectBridgeService/CacheTagsAdapterTest.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * @var CacheTagsAdapter
+     */
+    private $adapter;
+
+    /**
+     * @var MockObject
+     */
+    private $elementInterfaceMock;
+
```
Comment (JeroenVanDerLaan) [inline tests/unit/ObjectBridgeService/CacheTagsAdapterTest.php]: I think a test with multiple mocks should be added, as the adapter's method implies the use of multiple elements. I also think internally, `getCacheTags` does some array merging (assumption, didn't check). And I suspect clients expect a `$tags` array with all the merged cache tags from all given objects.

Correct me if I'm wrong.

File: src/ObjectBridgeService/CacheTagsAdapter.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+class CacheTagsAdapter
+{
+    /**
+     * Default functionality from ClassDefinition\Data\Object::getCacheTags
+     * This is a dummy and is mostly implemented by relation types
+     *
+     * @param ElementInterface $elementInterfaces
+     * @param array $tags
+     * @return array
+     */
```
Comment (JeroenVanDerLaan) [inline src/ObjectBridgeService/CacheTagsAdapter.php]: Small suggestion, I would go for a slightly different signature
`getMissingCacheTagsForElements(array $tags = [], ElementInterface ...$elements): array`.

In any case, the `: array` return type is missing.

File: tests/unit/ObjectBridgeService/CacheTagsAdapterTest.php (hunk 2 above)
Comment (Mitch391) [reply tests/unit/ObjectBridgeService/CacheTagsAdapterTest.php]: yeah that's right, but getCacheTags has to be mocked in the tests

**Pull request merged**